wp/core/Plugin.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Stakeborgdao\Badges;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Class Plugin
+ *
+ * @package Stakeborgdao\Badges
+ */
+class Plugin extends Singleton {
+
+	/**
+	 * Plugin constructor
+	 *
+	 * @param array $data
+	 */
+	public function __construct( array $data = [] ) {
+
+		parent::__construct();
+
+		add_action( 'init', array( $this, 'init_actions' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_assets' ) );
+
+	}
+
+	public function init_actions() {
+		add_shortcode( 'stakeborg_badges', array( $this, 'badges_shortcode' ) );
+	}
+
+	public function enqueue_assets() {
+		wp_register_style( 'stakeborg-badges', SBORG_BADGES_ASSETS_URL . 'build/main.css', array(), SBORG_BADGES_VERSION, 'all' );
+		wp_register_script( 'stakeborg-badges', SBORG_BADGES_ASSETS_URL . 'build/main.js', array(), SBORG_BADGES_VERSION, true );
```

Comment:
is it this that registers the 2 files? and thus gets them loaded?

Reply:
In WP you can register scripts and styles(using wp_register_style/script) but not actually loading them in the page. You can later use the handle used to register the scripts to actually enqueue them. This flow is more of a best practice that allows other devs to use the WP filters to dequeue maybe your script. 
https://developer.wordpress.org/reference/functions/wp_register_script/

Reply:
and yes, i meant registering them globally or whatever like a singleton. thanks :)